src/pages/home/report/ReportTypingIndicator.js

```diff
@@ -26,66 +26,54 @@ const defaultProps = {
     userTypingStatuses: {},
 };
 
-class ReportTypingIndicator extends React.Component {
-    constructor(props) {
-        super(props);
+const getUserTypingStatuses = (userTypingStatuses) => _.filter(_.keys(userTypingStatuses), (login) => userTypingStatuses[login]);
 
-        const usersTyping = props.userTypingStatuses ? _.filter(_.keys(props.userTypingStatuses), (login) => props.userTypingStatuses[login]) : [];
-        this.state = {usersTyping};
-    }
-
-    componentDidUpdate(prevProps) {
-        // Make sure we only update the state if there's been a change in who's typing.
-        if (_.isEqual(prevProps.userTypingStatuses, this.props.userTypingStatuses)) {
-            return;
-        }
+function ReportTypingIndicator(props) {
+    const [usersTyping, setUsersTyping] = useState(props.userTypingStatuses ? getUserTypingStatuses(props.userTypingStatuses) : []);
 
-        const usersTyping = _.filter(_.keys(this.props.userTypingStatuses), (login) => this.props.userTypingStatuses[login]);
+    useEffect(() => {
+        setUsersTyping(getUserTypingStatuses(props.userTypingStatuses));
+    }, [props.userTypingStatuses]);
```

Comment:
I know the original code was doing the similar thing

but do we need to create a separate local state for usersTyping?

can we do

```suggestion
   usersTyping = useMemo(() => {
        _.filter(_.keys(userTypingStatuses), (login) => userTypingStatuse[login]);
   }, [props.userTypingStatuses]);
```

Reply:
Yes @hayata-suenaga , we can do that; I think useMemo should be used here